lms/djangoapps/student_account/test/test_views.py

```diff
@@ -150,6 +150,35 @@ def test_password_change(self):
         result = self.client.login(username=self.USERNAME, password=self.NEW_PASSWORD)
         self.assertTrue(result)
 
+    def test_password_change_through_api(self):
+        # Log the user out
+        self.client.logout()
+
+        # Request a password change while logged out, simulating
+        # use of the password reset link from the login page
+        response = self._change_password_through_api(email=self.OLD_EMAIL)
+        self.assertEqual(response.status_code, 200)
+
+    def test_password_change_through_api_blank_email(self):
+        # Log the user out
+        self.client.logout()
+
+        # Request a password change while logged out, simulating
+        # use of the password reset link from the login page
+        # with blank email
+        response = self._change_password_through_api(email="")
+        self.assertEqual(response.status_code, 400)
+
+    def test_password_change_through_api_no_email(self):
+        # Log the user out
+        self.client.logout()
+
+        # Request a password change while logged out, simulating
+        # use of the password reset link from the login page
+        # with no email provided
+        response = self._change_password_through_api()
+        self.assertEqual(response.status_code, 400)
```

Comment:
In addition, I feel like you should add a test for the rate limit logic -- possibly by mocking out the limiter and pretending to have exceeded the limit, etc..